### DISCUSSION TITLE
Trivy to scan while docker build  not for already build images

### DISCUSSION THREAD

**Discussion by samkhamk** [aquasecurity/trivy#4049] [Q&A]
Hello Team,

We have pipeline which  does all the steps from cloning repo to build docker images, I wants to know how I can place Trivy their so that if any vulnerability is detected it should fails the build 

I know we can use Trivy command in Jenkins pipeline but I am not sure how Trivy will detect vulnerabilities while building docker images 

   stages{
        stage("Trivy Scan") {
            steps {
                script {
                    sh 'sudo curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sudo sh -s -- -b /usr/local/bin'
                    sh 'trivy image --light --no-progress --format template --template "@/<path-for-template>/html.tpl" -o /<path-for-jenkins>/workspace/$JOB_NAME/trivyscan.html alpine:3.10.1'
                    publishHTML([ allowMissing: false, alwaysLinkToLastBuild: false, keepAll: false, reportDir: '/<path-for-jenkins>/workspace/$JOB_NAME', reportFiles: 'trivyscan.html', reportName: 'TRIVY Scan Report', reportTitles: ''])
                }
            }
        }
    }
    
with above  example we can only scan already build image but what I want to use  Trivy such way that when docker build starts it detects vulnerabilities and fails build after detection 

can you please help me with examples or links to refer I will try that.

Regards,
samkhamk

**Comment by knqyf263** [comment]
The only approach Trivy currently provides is embedding Trivy into Dockerfile.
https://aquasecurity.github.io/trivy/v0.39/docs/advanced/container/embed-in-dockerfile/
